### ISSUE TITLE
Support "pre-release"/unstable metadata github releases

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#5286]
**What would you like Renovate to be able to do?**

Be able to determine if a "regular" versioned github release is in fact unstable, based on the metadata returned from GitHub's API. 

**Describe the solution you'd like**

The GitHub Releases datasource should look for the pre-release field per-release and return it as part of the datasource answer. Then, it should be used to determine "is unstable" during our lookup algorithm.

**Describe alternatives you've considered**

Currently we use version schemes to determine instability however it won't catch if a release "looks" stable, e.g. 2.0.0 but is considered a pre-release.

**Comment by HonkingGoose** [comment]
This is also a thing that the GitHub Actions updater part of Renovate will benefit from.

Example: `actions/setup-node` has a `2.1.2` [beta release](https://github.com/actions/setup-node/releases/tag/v2.1.2), so it would be better if Renovate would stay on the stable `1.4.4` [release](https://github.com/actions/setup-node/releases/tag/v1.4.4) instead of nudging users to take a unstable release.

**Comment by rarkins** [comment]
I looked into this yesterday and stopped once I discovered that `2.1.1` was tagged as non-prerelease in its metadata:

![image](https://user-images.githubusercontent.com/6311784/96835230-9760cd00-1443-11eb-9dac-f93fb3fe6b44.png)

Meaning even if I add code, Renovate will still propose 2.1.1 to users instead of 2.1.2, so it's not really going to achieve anything until GitHub fix things on their side.

The second challenge is that with actions we default to using github *tags*, because I believe there is no restriction that a tag must have an accompanying release and if we defaulted to releases then we'd potentially miss valid updates which are tag-only. I've created #7518 just now which should be implemented after this issue.
